### PR TITLE
refactor: rename `--list-all-pkgs` to `--scanners pkg`

### LINKED ISSUE #5183: Rename `--list-all-pkgs` to `--scanners sbom`
## Description
Currently, we have the `--scanners` flag, which supports the following options: vuln, misconf, secret, and license. To align with this structure, I'm considering renaming the `--list-all-pkgs` flag to` --scanners sbom`.

This proposed change aims to:

- Enhance the CLI's intuitiveness by grouping all scanner-related functionalities under the --scanners flag.
- Provide a more consistent user experience for those familiar with the --scanners flag.

However, I have a concern: while vuln, misconf, secret, and license are directly tied to security issues, the Software Bill of Materials (SBOM) is not directly a security concern. It's more about transparency and understanding the components within a software. By grouping it under the --scanners flag, there might be a potential for confusion among users regarding its purpose.

I'd appreciate feedback from the community on this proposal.

### PR DESCRIPTION
**Pull request by DmitriyLewen**
## Description
See #5183

TODO:
- [x] fix tests
- [x] update integration tests
- [x] update docs

## Related issues
- Close #5183

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: rpc/scanner/service.proto [hunk 1]
```diff
@@ -25,9 +25,8 @@ message Licenses {
 message ScanOptions {
   repeated string       vuln_type          = 1;
   repeated string       scanners           = 2;
-  bool                  list_all_packages  = 3;
-  map<string, Licenses> license_categories = 4;
-  bool                  include_dev_deps   = 5;
```
Comment (knqyf263) [inline rpc/scanner/service.proto:30]: The number should not be changed. We should use `reserve`.
https://protobuf.dev/programming-guides/proto3/#assigning

File: pkg/types/target.go [hunk 2]
```diff
@@ -43,6 +43,9 @@ const (
 
 	// LicenseScanner is the scanner of licenses
 	LicenseScanner = Scanner("license")
+
+	// SbomScanner includes all found packages on report
+	SbomScanner = Scanner("sbom")
```
Comment (knqyf263) [inline pkg/types/target.go:48]:
```suggestion
	SBOMScanner = Scanner("sbom")
```

File: rpc/scanner/service.proto (hunk 1 above)
Comment (DmitriyLewen) [reply rpc/scanner/service.proto]: I doubted this decision, but decided to think about it tomorrow. Thank you for saving me time!
Fixed in https://github.com/aquasecurity/trivy/pull/5255/commits/6b6a6cc9dabd301a85d912eb4d542081ee2ce8eb

File: pkg/types/target.go (hunk 2 above)
Comment (DmitriyLewen) [reply pkg/types/target.go]: Changed in https://github.com/aquasecurity/trivy/pull/5255/commits/6b6a6cc9dabd301a85d912eb4d542081ee2ce8eb

**Comment by knqyf263** [comment]
@DmitriyLewen Does this change make sense to you?

**Comment by DmitriyLewen** [comment]
When i started working on this PR - I wasn't sure we needed these changes.
Previously I used `--list-all-pkgs` flag. I got used to it, it was convenient for me and this flag was enough for me.

But when i created this PR i understood your opinion about this change.

We have scanners for vuln, license, secrets, etc... => we also need scanners to find installed packages.

But i had 1 problem. Many times i understood that i was confusing `sbom` scanner and `sbom` target. This can really confuse users.
Perhaps we can rename `--scanners sbom` to `--scanners packages` or something like that. This may help with tangling.

**Comment by knqyf263** [comment]
>We have scanners for vuln, license, secrets, etc... => we also need scanners to find installed packages.

Right. For example, there is a case where Trivy supports packages, but doesn't support vulnerabilities. It's easy for us to say `--scanner sbom` is supported, but `--scanner vuln` is not supported. It is more consistent.

>Perhaps we can rename --scanners sbom to --scanners packages or something like that. This may help with tangling.

I like it. Let's go with that idea. Thanks!

**Comment by DmitriyLewen** [comment]
> I like it. Let's go with that idea. Thanks!

Okay, i will change `sbom` to `packages` and write to you.

**Comment by knqyf263** [comment]
>Okay, i will change sbom to packages and write to you.

How about `--scanner pkg`?

**Comment by DmitriyLewen** [comment]
`--scanners pkgs` would be correct.
But I like. It's shorter than `packets`, but intuitive.

**Comment by DmitriyLewen** [comment]
@knqyf263  I renamed scanner. Take a look, when you have time, please

**Comment by knqyf263** [comment]
We need to rethink the design.

**Pull request closed without merging**